### ISSUE TITLE
ISCC.exe pre-processor arguments are broken in v5.5.6

### ISSUE THREAD

**Issue by extensia** [jrsoftware/issrc#172]
The ISCC.exe executable shipped in the new Inno Setup release fails when executed with pre-processor options.

Test Case

1) Open console window
2) CD into the Inno Setup install folder
3) Execute `ISCC.exe /pp-`

Expected Output:

```
Inno Setup 5 Command-Line Compiler
Copyright (C) 1997-2014 Jordan Russell. All rights reserved.
Portions Copyright (C) 2000-2014 Martijn Laan
Inno Setup Preprocessor
Copyright (C) 2001-2004 Alex Yackimoff. All rights reserved.

Script file name not specified.
```

Actual Output:

```
Copyright (C) 1997-2015 Jordan Russell. All rights reserved.
Portions Copyright (C) 2000-2015 Martijn Laan
Inno Setup Preprocessor
Copyright (C) 2001-2004 Alex Yackimoff. All rights reserved.

Unknown option: /pp-
```

This breakage appears to be related to the problem that occurred with the initial v5.5.5 release, but was fixed in v5.5.5a. Please see my earlier report for additional details.

Regards,
David

**Comment by martijnlaan** [comment]
What do you mean with 5.5.5a? And what earlier report? Thanks.

**Comment by martijnlaan** [comment]
I checked and /pp- never worked before due to a bug. You just didn't get an error, instead it was ignored. As a workaround use `#pragma parseroption -p-` in the script.

**Comment by extensia** [comment]
Sorry. For some reason I didn't receive a notification when you posted your response.

It is entirely possible that the /p switch wasn't working correctly in earlier builds. The switch value I set is in fact the ISPP default. The reason it was being set was to ensure the string mode was correct when calling Inno Setup from a Gradle build plugin I developed. I needed to escape quotes in strings being passed from Gradle and needed to know the correct escaping mode was enabled.

My comment about the version 5.5.5 build is related to issue #170 which was prematurely marked as closed. There was clearly a problem with the v5.5.5 ISCC.exe binary that was checked into the ispack GitHub repo. Not sure exactly what happened, but that binary was broken and caused major  headaches until I finally tracked down the problem.

I am currently using the binaries installed by ispack-5.5.5.exe and those work correctly. By correctly I mean the ISCC.exe program accepts all of the documented switches I pass and behaves as expected. I tried upgrading to v5.5.6, but was unable to use the new version due to the breakage I noted in this issue. Not sure if your commit to address the /p switch fixes all of the problems. I would be happy to test a new build if you would be kind enough to create one from the new 5.5.7 version.

Best Regards,
David
